From 3d51bb5e819bbeb023fcee59e402eee96bc232b0 Mon Sep 17 00:00:00 2001
From: Aaron Fabbri <fabbri@cloudera.com>
Date: Thu, 15 Sep 2016 16:51:42 -0700
Subject: [PATCH 1859/2748] CLOUDERA-BUILD CDH-43922 TestIntegration failures
 due to staging dir location

Our staging dir defaults differ from upstream which was causing this test to
fail.  The different default is due to a cloudera-only MR1 commit
8abf9b437e5131d6aeff8e424bfe9acba11cb8e0

Change-Id: I107e11e26f1c208dc90c0c5e181cbe9d3a725979
---
 .../org/apache/hadoop/tools/TestDistCpViewFs.java  |    4 ++++
 .../org/apache/hadoop/tools/TestIntegration.java   |    4 ++++
 2 files changed, 8 insertions(+)

diff --git a/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestDistCpViewFs.java b/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestDistCpViewFs.java
index a6939a2..e903ffa 100644
--- a/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestDistCpViewFs.java
+++ b/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestDistCpViewFs.java
@@ -24,6 +24,7 @@
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.server.jobtracker.JTConfig;
 import org.apache.hadoop.tools.util.TestDistCpUtils;
 import org.apache.hadoop.fs.FsConstants;
 
@@ -49,6 +50,9 @@ private static Configuration getConf() throws URISyntaxException {
     Configuration conf = new Configuration();
     conf.set("mapred.job.tracker", "local");
     conf.set("fs.default.name", "file:///");
+    // CLOUDERA-BUILD work around our different default for staging dir which
+    // breaks many of these tests. See CDH-43922.
+    conf.set(JTConfig.JT_STAGING_AREA_ROOT, "/tmp/hadoop/mapred/staging");
     return conf;
   }
 
diff --git a/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestIntegration.java b/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestIntegration.java
index ee8e7cc..10d87f4 100644
--- a/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestIntegration.java
+++ b/hadoop-tools/hadoop-distcp/src/test/java/org/apache/hadoop/tools/TestIntegration.java
@@ -26,6 +26,7 @@
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.mapreduce.Cluster;
 import org.apache.hadoop.mapreduce.JobSubmissionFiles;
+import org.apache.hadoop.mapreduce.server.jobtracker.JTConfig;
 import org.apache.hadoop.security.Credentials;
 import org.apache.hadoop.tools.util.TestDistCpUtils;
 import org.junit.Assert;
@@ -67,6 +68,9 @@ private static Configuration getConf() {
     Configuration conf = new Configuration();
     conf.set("fs.default.name", "file:///");
     conf.set("mapred.job.tracker", "local");
+    // CLOUDERA-BUILD work around our different default for staging dir which
+    // breaks many of these tests. See CDH-43922.
+    conf.set(JTConfig.JT_STAGING_AREA_ROOT, "/tmp/hadoop/mapred/staging");
     return conf;
   }
 
-- 
1.7.9.5

